### ISSUE TITLE
(inofficial) automatic yearly system upgrade script

### ISSUE THREAD

**Issue by Konfekt** [yast/yast-online-update-configuration#25]
To ease maintenance for less experienced users, I put [online](https://github.com/Konfekt/opensuse-distribution-upgrade.sh/) simple instructions to run a simple shell script that runs `zypper distribution-upgrade` with options that automate the process as much as possible.
Since most users' needs are covered by installing the software available in the official repositories, it usually suffices to pass the increased `$releasever` to `zypper`.
Could a similar unofficial script be deployed by openSUSE, similar to what other distributions such as Ubuntu, offer?
(In contrast to the linked script, such a script would then not update the packman repositories due to license restrictions.)

**Comment by ancorgs** [comment]
This is also reported at bugzilla here: https://bugzilla.suse.com/show_bug.cgi?id=1178004

To avoid duplication, I'm going to close this github issue since bugzilla is the main tracking platform for core YaST modules.

**Comment by ancorgs** [comment]
I closed this by mistake with a wrong comment I have already deleted. Sorry for the noise.

**Comment by ancorgs** [comment]
Maybe interesting for @Konfekt  https://hackweek.suse.com/projects/easy-opensuse-upgrade

**Comment by Konfekt** [comment]
Thank you, I knew not of [these prior discussions](https://progress.opensuse.org/issues/20910) of this ancient issue, whose resolution seems to be still pending.

**Comment by maverick74** [comment]
> Could a similar unofficial script be deployed by openSUSE

Why "_unofficial_"?
Why can't openSUSE provide a nice and simple GUI (like in Ubuntu...) that alerts the user of new versions and upgrades the system with a simple "Yes"+Root_Password as an official package bundled by default?
Is there any legal problem i'm not aware of?

**Comment by Konfekt** [comment]
Inertia: nobody did the work; mine is sufficient for install and forget users, but if there are many custom repos, trouble looms.

**Comment by maverick74** [comment]
yes, but... most users for sure only use the official repos (maybe packman). 
So I believe that the cases where custom repos exists are minimal and where those exist their users have the know-how how to fix the problems.

We could also play the safe side and have the option off by default until we have better clue of "trouble probabilities"

**Comment by Konfekt** [comment]
> as soon as i tell anyone "open the terminal" -> that's where i loose them!)

True, though my instructions make an effort to achieve that anyways.

**Comment by maverick74** [comment]
There's nothing wrong with your approach!!!
As there is nothing wrong with opening the terminal and just type `zypper --releasever=[new_release_number] dup `!
There is also a lot of other solutions seen on the link you shared.
Most (if not all) are plain simple!!!

Still, we have the fact that users run away when anyone speaks the T-word
Most outside (suse) developers eventually drop their projects (hence having an _official_ package would be good). 
Having it done officially would also be good because it would be done "their way" as - it seems to **_me_** - suse is not very open to outside contributions (i don't recall correctly which one was, but there was a project that was expected to be picked up by suse that "never was" - i don't know why but there seems to be very little interest in this from suse)

All in all i don't see this as taking much time to be done, neither to be maintained... i think it's more a question of "will".

I think it's sad because openSUSE is, from my POV, the most advanced and stable distro and comparing it to other big OS's... (Windows, macOS, Ubuntu and i believe others) everyone is doing it...
yes... occasionally it breaks down - same happens when i accidentally spill my coffee over the computer - and i just keep drinking it :p
(this got a little big, boring, "personal" and maybe irrelevant... so i'm ok with mods hidding my comment)

**Comment by maverick74** [comment]
I've been "ruminating" on this... 
@ancorgs can you give us some kind of reason why suse doesn't implement something like this?

it seems something really fast&easy to do (to a developer, of course...) and it's something that would benefit greatly less experienced users! It puzzles me how something like this has been in the wait list (if it's even there...) so many years

**Comment by jreidinger** [comment]
module dropped. Sorry

**Comment by maverick74** [comment]
Thanks for the update @jreidinger ! 🙂

I was expecting it, considering the direction of the project regarding Yast

Would be good if Discover could be used to perform the upgrades (as it does on fedora) ...
